tests/daemon/test-persistent-cache.c

```diff
@@ -349,1064 +302,449 @@ boot_timestamp_is_valid (gint64 previous_relative_time,
-                                                      &capacity));
-
-  // An empty cache should be in the LOW capacity state.
-  g_assert_cmpint (capacity, ==, CAPACITY_LOW);
-  g_assert_cmpint (num_singulars_stored, ==, 0);
-  g_assert_cmpint (num_aggregates_stored, ==, 0);
-  g_assert_cmpint (num_sequences_stored, ==, 0);
+  guint64 token = assert_variants_read (cache, variants, num_variants);
+  GError *error = NULL;
+  gboolean remove_succeeded =
+    emer_persistent_cache_remove (cache, token, &error);
 
-  g_object_unref (cache);
+  g_assert_no_error (error);
+  g_assert_true (remove_succeeded);
+  assert_cache_is_empty (cache);
 }
 
-static void
-test_persistent_cache_store_one_singular_event_succeeds (gboolean     *unused,
-                                                         gconstpointer dontuseme)
-{
-  EmerPersistentCache *cache = make_testing_cache ();
-  capacity_t capacity;
-  g_assert_true (store_single_singular_event (cache, &capacity));
-  g_object_unref (cache);
-  g_assert_cmpint (capacity, ==, CAPACITY_LOW);
-}
+/* ----- Actual Test Cases below ------ */
 
 static void
-test_persistent_cache_store_one_aggregate_event_succeeds (gboolean     *unused,
-                                                          gconstpointer dontuseme)
+test_persistent_cache_new (gboolean     *unused,
+                           gconstpointer dontuseme)
 {
   EmerPersistentCache *cache = make_testing_cache ();
-  capacity_t capacity;
-  g_assert_true (store_single_aggregate_event (cache, &capacity));
+  g_assert_nonnull (cache);
   g_object_unref (cache);
-  g_assert_cmpint (capacity, ==, CAPACITY_LOW);
 }
 
 static void
-test_persistent_cache_store_one_sequence_event_succeeds (gboolean     *unused,
-                                                         gconstpointer dontuseme)
+test_persistent_cache_store_none (gboolean     *unused,
+                                  gconstpointer dontuseme)
 {
   EmerPersistentCache *cache = make_testing_cache ();
-  capacity_t capacity;
-  g_assert_true (store_single_sequence_event (cache, &capacity));
+
+  assert_variants_stored (cache, NULL /* variants */, 0);
+
   g_object_unref (cache);
-  g_assert_cmpint (capacity, ==, CAPACITY_LOW);
 }
 
 static void
-test_persistent_cache_store_one_of_each_succeeds (gboolean     *unused,
-                                                  gconstpointer dontuseme)
+test_persistent_cache_store_one (gboolean     *unused,
+                                 gconstpointer dontuseme)
 {
   EmerPersistentCache *cache = make_testing_cache ();
 
-  SingularEvent singular_array[1];
-  make_singular_event (0, singular_array);
-
-  AggregateEvent aggregate_array[1];
-  make_aggregate_event (0, aggregate_array);
-
-  SequenceEvent sequence_array[1];
-  make_sequence_event (0, sequence_array);
+  GVariant *variants[] = { make_variant (0) };
+  gsize num_variants = G_N_ELEMENTS (variants);
+  assert_variants_stored (cache, variants, num_variants);
```

Comment:
That actually causes a crash because `assert_variants_stored` consumes the floating reference.